### ISSUE TITLE
UnboundLocalError on django-reversion historical view of page

### ISSUE THREAD

**Issue by 747project** [divio/django-cms#3815]
In `django-cms 2.4.3`, when using the package `django-reversion`, if no placeholders match a particular reversion history page view, then the variable `placeholder` is not set.

Error received when trying to view any version of a cms page where no matching placeholders are found:

```
Traceback:
File "/home/username/virtualenvs/projectname/local/lib/python2.7/site-packages/django/core/handlers/base.py" in get_response
  113.                         response = callback(request, *callback_args, **callback_kwargs)
File "/home/username/virtualenvs/projectname/local/lib/python2.7/site-packages/django/utils/decorators.py" in _wrapped_view
  91.                     response = view_func(request, *args, **kwargs)
File "/home/username/virtualenvs/projectname/local/lib/python2.7/site-packages/django/views/decorators/cache.py" in _wrapped_view_func
  89.         response = view_func(request, *args, **kwargs)
File "/home/username/virtualenvs/projectname/local/lib/python2.7/site-packages/django/contrib/admin/sites.py" in inner
  202.             return view(request, *args, **kwargs)
File "/home/username/virtualenvs/projectname/local/lib/python2.7/site-packages/reversion/revisions.py" in do_revision_context
  301.                     return func(*args, **kwargs)
File "/home/username/virtualenvs/projectname/local/lib/python2.7/site-packages/cms/admin/pageadmin.py" in revision_view
  714.         response = super(PageAdmin, self).revision_view(request, object_id, version_id, extra_context)
File "/home/username/virtualenvs/projectname/local/lib/python2.7/site-packages/django/db/transaction.py" in inner
  223.                 return func(*args, **kwargs)
File "/home/username/virtualenvs/projectname/local/lib/python2.7/site-packages/reversion/admin.py" in revision_view
  409.         return self.render_revision_form(request, obj, version, context, revert=True)
File "/home/username/virtualenvs/projectname/local/lib/python2.7/site-packages/cms/admin/pageadmin.py" in render_revision_form
  738.         return super(PageAdmin, self).render_revision_form(request, obj, version, context, revert, recover)
File "/home/username/virtualenvs/projectname/local/lib/python2.7/site-packages/reversion/admin.py" in render_revision_form
  264.         ModelForm = self.get_form(request, obj)
File "/home/username/virtualenvs/projectname/local/lib/python2.7/site-packages/cms/admin/pageadmin.py" in get_form
  435.                     'placeholder': placeholder

Exception Type: UnboundLocalError at /admin/cms/page/22/history/1523/
Exception Value: local variable 'placeholder' referenced before assignment
```

**Comment by evildmp** [comment]
I think this is going to be a low priority for the project. I have marked the issue as Accepted, but in practice I think it is unlikely that the core team is going to find the time to tackle it. We are happy to receive patches for it of course though.

**Comment by 747project** [comment]
Thank you @evildmp for considering this. A pull request has been submitted, which fixes this issue, as well as #3814.

**Comment by czpython** [comment]
Fixed by #3805
